daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/LayeredSequenceParser.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.processors.parsers
+
+import org.apache.daffodil.processors.TermRuntimeData
+import org.apache.daffodil.processors.LayerTransformerEv
+import org.apache.daffodil.io.ByteBufferDataInputStream
+
+class LayeredSequenceParser(rd: TermRuntimeData,
+  layerTransformerEv: LayerTransformerEv,
+  bodyParser: Parser)
+  extends SequenceCombinatorParser(rd, bodyParser) {
```

Comment:
Ah, so maybe this solves the problem I mentioned elsewhere about duplicating sequence combinator logic.

Interestingly though, does this mean that the sequence combinator will actually used properties like dfdl:separatorSuppresstionPolicy from the layered sequence (assuming is was provided in the default format)?

Reply:
Interesting point. I should at least document whether LayeredSequenceParser overrides everything about SequenceCombinatorParser that uses the other non-layering properties, but in that case, why derive from it. I'll give this some thought and at minimum add doc.

Reply:
Turns out there is no good reason to inherit from SequenceCombinatorParser, so I'm eliminating that here and for the LayeredSequenceUnparser similarly.

Reply:
So both the parser and unparser both call super.parse(state) or super.unparse(state) so as to reuse the guts of the parser.

Once the parser and unparser for Sequence are revised though, this may no longer make sense.  There is a lot of code reuse for unparser especially here. No matter what happens when Sequence is reimplemented this code reuse will need to be preserved. 

The answer to your question is no however, the properties like dfdl:separatorSuppressionPolicy will not apply, as the body parser for these cannot have an array (now enforced), and can be only one term, and the grammar doesn't put down separator grammar terms around it.  So it will not have any "sequence-like" behavior associated with separators. (Unless of course the single body itself IS a sequence, but then the properties are its own, not those of the layered sequence.)